frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,153 @@
-import React from 'react';
+import React, {useState} from 'react';
+import {Box, Button, Checkbox, Link, TextField, Typography} from '@mui/material';
+import {Paths} from "../../../common/constansts/paths";
 import {Titles} from "../../../common/constansts/titles";
+import Logo from "../../../img/Logo.png";
+import "../../../common/styles/button.css";
+import "./styles.css";
+import {useNavigate} from "react-router-dom";
+import {UserCredentials} from "../../../types/userCredentials";
+import {Buttons} from "../../../common/constansts/buttons";
+import {Messages} from "../../../common/constansts/messages";
+import {Roles} from "../../../common/constansts/roles";
+import {UserType} from "../../../types/userType";
 
 const SignUp: React.FC = () => {
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [name, setName] = useState('');
+    const navigate = useNavigate();
+    const [customerChecked, setCustomerChecked] = useState(false);
+    const [merchantChecked, setMerchantChecked] = useState(false);
+    const [agreeChecked, setAgreeChecked] = useState(false);
+
+    const handleCustomerCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setCustomerChecked(event.target.checked);
+        if (event.target.checked) {
+            setMerchantChecked(false);
+        }
+    };
+
+    const handleMerchantCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setMerchantChecked(event.target.checked);
+        if (event.target.checked) {
+            setCustomerChecked(false);
+        }
+    };
+
+    const handleAgreeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setAgreeChecked(event.target.checked);
+    };
+
+    const handleSignUp = () => {
+        const isSignUpEnabled = email && name && password && (customerChecked || merchantChecked) && agreeChecked;
+        if (isSignUpEnabled) {
+            const userCredentials = createUserCredentials(email, name, password);
+            const userType = getUserType(customerChecked);
+            console.log("userCredentials: " + userCredentials + ", userType: " + userType);
+            navigate(Paths.SignIn);
+        }
+    };
+
+    const createUserCredentials = (name: string, email: string, password: string): UserCredentials => {
+        return {
+            email,
+            name,
+            password
+        }
+    }
+
+    const getUserType = (customerChecked: boolean): UserType => {
+        if (customerChecked) {
+            return UserType.Customer;
+        }
+        return UserType.Merchant;
+    }
+
     return (
-        <p>
-            {Titles.SIGN_UP}
-        </p>
+        <Box className="box">
+            <Box className="box-inner">
+                <Box
+                    component="img"
+                    className="logo"
+                    alt={Titles.APP_NAME}
+                    src={Logo}
+                />
```

Comment:
you are going to use the same icon component for sign in, so it's better to extract it